src/popup/components/Modals/TransferReceive.vue

```diff
@@ -101,95 +115,102 @@ export default {
+    const availableTokens = computed<ITokenList>(
+      () => root.$store.state.fungibleTokens.availableTokens,
+    );
+    const getAeternityToken = computed(() => root.$store.getters['fungibleTokens/getAeternityToken']);
+
+    const { copied, copy } = useCopy();
+
+    function getTokenInfo(link = false) {
+      if (amount.value <= 0) return '';
+      const token = selectedAsset && (selectedAsset.value?.symbol === 'AE'
         ? 'AE'
-        : this.selectedAsset.contractId);
+        : selectedAsset.value?.contractId);
       const firstChar = link ? '&' : '?';
       const separator = link ? '&' : '&amp;';
-      return `${firstChar}token=${token}${separator}amount=${this.amount}`;
-    },
-    getLink(value) {
-      return `${APP_LINK_WEB}/account?account=${value}${this.getTokenInfo(true)}`;
-    },
-    async share() {
-      const { address } = this.account;
-      const walletLink = this.getLink(address);
-      const text = (this.amount > 0)
-        ? this.$t('modals.receive.shareTextNoAmount', { address, walletLink })
-        : this.$t('modals.receive.shareTextWithAmount', { address, walletLink, amount: this.amount });
-      await this.$store.dispatch('share', { text });
-    },
-    getQRdata(address) {
-      return this.amount > 0
-        ? `${this.getLink(address)}`
-        : address;
-    },
-    getTextToCopy() {
-      return this.amount > 0
-        ? this.getLink(this.account.address)
-        : this.account.address;
-    },
-    handleAssetChange(asset) {
-      this.selectedAsset = asset;
-    },
-    closeModal() {
-      this.$store.commit('modals/closeByKey', MODAL_TRANSFER_RECEIVE);
-    },
+      return `${firstChar}token=${token}${separator}amount=${amount.value}`;
+    }
+
+    function getLink(value: string) {
+      return `${APP_LINK_WEB}/account?account=${value}${getTokenInfo(true)}`;
+    }
+
+    const accountAddressAsLink = computed(() => {
+      return (amount.value > 0)
+        ? getLink(account.value.address)
+        : account.value.address;
+    });
+
+    const computedAddress = computed(() => {
+      return (amount.value > 0)
+        ? `${account.value.address}?${getTokenInfo(true).substring(1)}`
+        : account.value.address;
+    });
+
+    async function share() {
+      const { address } = account.value;
+      const walletLink = getLink(address);
+      const text = (amount.value > 0)
+        ? i18n.t('modals.receive.shareTextNoAmount', { address, walletLink })
+        : i18n.t('modals.receive.shareTextWithAmount', { address, walletLink, amount: amount.value });
+      await root.$store.dispatch('share', { text });
+    }
+
+    function handleAssetChange(asset: IAsset) {
+      selectedAsset.value = asset;
+    }
+
+    function closeModal() {
+      root.$store.commit('modals/closeByKey', MODAL_TRANSFER_RECEIVE);
+    }
+
+    function copyAddress() {
+      copy(accountAddressAsLink.value);
+    }
+
+    (() => {
```

Comment:
onMounted hook will be better

Reply:
Why?

Reply:
I'm worried about that option, because vue works on hook, and give us methods of lifecycle. I never saw on documentation ify

Reply:
@pstachel what do you think about that?

Reply:
Yes, I think we should stick to the lifecycle methods. Also I think the component readability will increase

Reply:
After all no hook needed.